src/DataLakeStore/DataLakeStore/DataPlaneModels/AdlsClientFactory.cs

```diff
@@ -43,25 +45,16 @@ internal static AdlsClient GetAdlsClient(string accntNm, IAzureContext context)
                 {
                     return ClientFactory[accntNm];
                 }
-                ServiceClientCredentials creds;
+                AdlsClient client = null;
```

Comment:
How does IsTest work?  Generally there should not be any dependency on test code in shipped cmdlets.  At least this needs to be protected by a #if for DEBUG versus Release builds.  Ideally you would use a Provider (We provide a registry for factories in AzureSession), so the factory can be provided in tests, but has a default value in the code.

Reply:
I agree provider is the best way, however I cannot directly use ICLientFactory (AzureSession.Instance.ClientFactory) to generate my client in a generic way because my client does not inherit from ServiceClient. but I agree way we can think of a better way to do this. I will create an issue related to this, but I dont think this will be a blocking issue.

 For now here is what we can do. I willmake this IsTest variables internal and use it only in "internalvisible" way to run only in my tests. In that way we do not expose these test specific stuff for use. 